test/common/http/http1/codec_impl_test.cc

```diff
@@ -2238,12 +2238,23 @@ TEST_P(Http1ClientConnectionImplTest, BadEncodeParams) {
 
   NiceMock<MockResponseDecoder> response_decoder;
 
-  // Need to set :method and :path
+  // Need to set :method and :path.
+  // New and legacy codecs will behave differently on errors from processing outbound data. The
+  // legacy codecs will throw an exception (that presently will be uncaught in contexts like
+  // sendLocalReply), while the new codecs temporarily RELEASE_ASSERT until Envoy handles errors on
+  // outgoing data.
   Http::RequestEncoder& request_encoder = codec_->newStream(response_decoder);
-  EXPECT_THROW(request_encoder.encodeHeaders(TestRequestHeaderMapImpl{{":path", "/"}}, true),
-               CodecClientException);
-  EXPECT_THROW(request_encoder.encodeHeaders(TestRequestHeaderMapImpl{{":method", "GET"}}, true),
-               CodecClientException);
+  if (GetParam()) {
```

Comment:
nit: could you wrap this GetParam in a accessor method that has a more descriptive name?

Reply:
all set!